### ISSUE TITLE
aria-active does not work for switch-button 

### ISSUE THREAD

**Issue by teomaragakis** [teomaragakis/polymer-knobs#16]

**Comment by ktiedt** [comment]
Is still still valid for press-button after the latest rounds of tweaks? I believe so.

**Comment by teomaragakis** [comment]
It still does not work, but I'm starting to think that there's a logical issue here. Toggle buttons act as checkboxes and thus should have the role="checkbox" attribute. That does not apply to auto-release buttons though. Aria attributes follow (and depend on) an element with a particular role. Maybe we should keep the two buttons in separate files as it was originally?

See: https://github.com/teomaragakis/polymer-knobs/commit/5f2e45a004316fb20f6331707c3d9420a1017157#diff-f65c5bc4d0798b79ef69c6d2582f9a34L163

**Comment by ktiedt** [comment]
I think it will be fine, because even a momentary button should report its state change while the spacebar/enter key is pressed.

**Comment by ktiedt** [comment]
We can still break this out if you want, however this will now work for toggleable buttons... if we do want to report aria state for momentary buttons I think we could add a very small block of code that basically does:
```javascript
// on-pressed handler
if (!this.toggles) { this.active = this.pressed; }
``
and I think that would satisfy the aria code and make it report the pressed state for that as well.

**Comment by teomaragakis** [comment]
I think it makes more sense that the toggle button remains with `role="checkbox"`, and that the momentary one assumes a button role? This way one can use aria-checked and the other aria-pressed. Is there a way to add hostAttributes programmatically?

**Comment by ktiedt** [comment]
Valid point, and I am not sure re: modifying hostAttributes on the fly

**Comment by davidrissato** [comment]
I'm in vacation so I haven't seen your code yet, but Teo's question drew my attention. I don't know if you were asking that, but if you need to change host attributes for a custom property you can simply declare a property in your component and set it with 'reflectToAttribute: true'. If ot isn't a custom property (lile aria-*), you can use 'this.toggleAttribute("attribute-name", boolEnabled)' to change it programatically. Many components in polymer libraries use those technics to control display behaviors, in conjunction to CSS atrribute selectors, like ':host[aria-active] div.something { background-color: green; }'.

**Comment by davidrissato** [comment]
Ps: sorry for typos. I'm typing from my phone :-)

**Comment by teomaragakis** [comment]
Hey, @davidrissato, I know that, but iron and paper elements do aria attributes in the `hostAttributes` section. Just enjoy as much holiday as you have left and we'll have a nice chat about that when you're back.
